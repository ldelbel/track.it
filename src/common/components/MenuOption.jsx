import React, { useState, useEffect } from 'react'
import { useHistory } from 'react-router-dom';
import PropTypes from 'prop-types'
import { AiOutlinePlus, AiOutlineHistory, AiFillPieChart } from 'react-icons/ai';
import { FaRunning } from 'react-icons/fa';
import { RiMoreFill } from 'react-icons/ri';
import { IconContext } from 'react-icons';
import styles from '../styles/MenuOption.module.scss';


const MenuOption = props => {
  const { page, selected , handleUpdateSelected } = props;
  const [style, setStyle] = useState();
  let history = useHistory();
   
  useEffect(()=>{
    selected ? (
      setStyle(styles.container_selected)
    ) : (
      setStyle(styles.container_notselected)
    )
  }, [selected])

  const handleClick = () => {
    handleUpdateSelected(page);
    page === 'new' ? (
      history.push(`/app/`)
    ) : (
      history.push(`/app/${page}`)
    )
  }

  let content = (
    <>
      <IconContext.Provider value={{ 'size': '0.6em'}}>
        <AiOutlinePlus /> 
      </IconContext.Provider>          
      <FaRunning />
    </>
  );

  let name = '';

  switch(page) {
    case 'new':
      content = (
        <>
          <IconContext.Provider value={{ 'size': '0.6em'}}>
            <AiOutlinePlus /> 
          </IconContext.Provider>          
          <FaRunning />
        </>
      );
      name = 'New Run'
      break;
    case 'history':
      content = <AiOutlineHistory />;
      name = 'History';
      break;
    case 'progress':
      content = <AiFillPieChart />;
      name = 'Your Progress';
      break;
    case 'more':
      content = <RiMoreFill />;
      name = 'More';
      break;
    default:
  }
    
  return (
    <button
    className={style}
    onClick={handleClick}>
      <div>
        <span>
          {content}
        </span>
        <p>{name}</p>
      </div>
    </button>
  );
}

MenuOption.propTypes = {

}

export default MenuOption;
