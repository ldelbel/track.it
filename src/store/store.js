import { createStore } from 'redux';
import menuReducer from '../reducers/menuReducer';

const store = createStore(menuReducer);

export default store;