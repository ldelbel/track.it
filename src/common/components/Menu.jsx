import React, { useState } from 'react';
import styles from '../styles/Menu.module.scss';
import MenuOption from './MenuOption';

const Menu = () => {
  const [selected, setSelected] = useState({
    new: true,
    history: false,
    progress: false,
    more: false,
  });

  const defValue = (keyToCompare, keyReference ,value) => {
    return keyToCompare === keyReference ? true : false
  }

  const toggle = (obj, key, fn) => (
    Object.fromEntries(
      Object.entries(obj).map(
        ([k, v]) => [k, fn(k,key,v)]
      )
    )
  )

  const handleUpdateSelected = (key) => {
    setSelected(toggle(selected,key, defValue))
  }

  return (
    <div className={styles.container}>
      <MenuOption page={'new'} selected={selected.new} handleUpdateSelected={handleUpdateSelected} />
      <MenuOption page={'history'} selected={selected.history} handleUpdateSelected={handleUpdateSelected} />
      <MenuOption page={'progress'} selected={selected.progress} handleUpdateSelected={handleUpdateSelected} />
      <MenuOption page={'more'} selected={selected.more} handleUpdateSelected={handleUpdateSelected} />
    </div>
  );
}

export default Menu;