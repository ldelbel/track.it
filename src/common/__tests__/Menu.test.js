import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow } from 'enzyme';
import Menu from '../components/Menu';
import MenuOption from '../components/MenuOption';
import '@testing-library/jest-dom/extend-expect';
import store from '../../store/store';

configure({ adapter: new Adapter() });

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => ({
    pathname: '/app/',
  }),
}));

describe('menu component', () => {
  it('renders correctly', () => {
    const menu = renderer.create(<Menu />).toJSON();
    expect(menu).toMatchSnapshot();
  });

  it('contains MenuOptions as children', () => {
    const wrapper = shallow(<Menu store={store} />);
    expect(wrapper.containsMatchingElement(<MenuOption />)).toEqual(true);
  });
});
