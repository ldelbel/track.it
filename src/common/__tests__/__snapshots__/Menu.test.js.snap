// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`menu component renders correctly 1`] = `
<div
  className="container"
>
  <button
    onClick={[Function]}
  >
    <div>
      <span>
        <svg
          fill="currentColor"
          height="0.6em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          t="1551322312294"
          version="1.1"
          viewBox="0 0 1024 1024"
          width="0.6em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs />
          <path
            d="M474 152m8 0l60 0q8 0 8 8l0 704q0 8-8 8l-60 0q-8 0-8-8l0-704q0-8 8-8Z"
          />
          <path
            d="M168 474m8 0l672 0q8 0 8 8l0 60q0 8-8 8l-672 0q-8 0-8-8l0-60q0-8 8-8Z"
          />
        </svg>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 416 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M272 96c26.51 0 48-21.49 48-48S298.51 0 272 0s-48 21.49-48 48 21.49 48 48 48zM113.69 317.47l-14.8 34.52H32c-17.67 0-32 14.33-32 32s14.33 32 32 32h77.45c19.25 0 36.58-11.44 44.11-29.09l8.79-20.52-10.67-6.3c-17.32-10.23-30.06-25.37-37.99-42.61zM384 223.99h-44.03l-26.06-53.25c-12.5-25.55-35.45-44.23-61.78-50.94l-71.08-21.14c-28.3-6.8-57.77-.55-80.84 17.14l-39.67 30.41c-14.03 10.75-16.69 30.83-5.92 44.86s30.84 16.66 44.86 5.92l39.69-30.41c7.67-5.89 17.44-8 25.27-6.14l14.7 4.37-37.46 87.39c-12.62 29.48-1.31 64.01 26.3 80.31l84.98 50.17-27.47 87.73c-5.28 16.86 4.11 34.81 20.97 40.09 3.19 1 6.41 1.48 9.58 1.48 13.61 0 26.23-8.77 30.52-22.45l31.64-101.06c5.91-20.77-2.89-43.08-21.64-54.39l-61.24-36.14 31.31-78.28 20.27 41.43c8 16.34 24.92 26.89 43.11 26.89H384c17.67 0 32-14.33 32-32s-14.33-31.99-32-31.99z"
          />
        </svg>
      </span>
      <p>
        New Run
      </p>
    </div>
  </button>
  <button
    onClick={[Function]}
  >
    <div>
      <span>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M536.1 273H488c-4.4 0-8 3.6-8 8v275.3c0 2.6 1.2 5 3.3 6.5l165.3 120.7c3.6 2.6 8.6 1.9 11.2-1.7l28.6-39c2.7-3.7 1.9-8.7-1.7-11.2L544.1 528.5V281c0-4.4-3.6-8-8-8zm219.8 75.2l156.8 38.3c5 1.2 9.9-2.6 9.9-7.7l.8-161.5c0-6.7-7.7-10.5-12.9-6.3L752.9 334.1a8 8 0 0 0 3 14.1zm167.7 301.1l-56.7-19.5a8 8 0 0 0-10.1 4.8c-1.9 5.1-3.9 10.1-6 15.1-17.8 42.1-43.3 80-75.9 112.5a353 353 0 0 1-112.5 75.9 352.18 352.18 0 0 1-137.7 27.8c-47.8 0-94.1-9.3-137.7-27.8a353 353 0 0 1-112.5-75.9c-32.5-32.5-58-70.4-75.9-112.5A353.44 353.44 0 0 1 171 512c0-47.8 9.3-94.2 27.8-137.8 17.8-42.1 43.3-80 75.9-112.5a353 353 0 0 1 112.5-75.9C430.6 167.3 477 158 524.8 158s94.1 9.3 137.7 27.8A353 353 0 0 1 775 261.7c10.2 10.3 19.8 21 28.6 32.3l59.8-46.8C784.7 146.6 662.2 81.9 524.6 82 285 82.1 92.6 276.7 95 516.4 97.4 751.9 288.9 942 524.8 942c185.5 0 343.5-117.6 403.7-282.3 1.5-4.2-.7-8.9-4.9-10.4z"
          />
        </svg>
      </span>
      <p>
        History
      </p>
    </div>
  </button>
  <button
    onClick={[Function]}
  >
    <div>
      <span>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M863.1 518.5H505.5V160.9c0-4.4-3.6-8-8-8h-26a398.57 398.57 0 0 0-282.5 117 397.47 397.47 0 0 0-85.6 127C82.6 446.2 72 498.5 72 552.5S82.6 658.7 103.4 708c20.1 47.5 48.9 90.3 85.6 127 36.7 36.7 79.4 65.5 127 85.6a396.64 396.64 0 0 0 155.6 31.5 398.57 398.57 0 0 0 282.5-117c36.7-36.7 65.5-79.4 85.6-127a396.64 396.64 0 0 0 31.5-155.6v-26c-.1-4.4-3.7-8-8.1-8zM951 463l-2.6-28.2c-8.5-92-49.3-178.8-115.1-244.3A398.5 398.5 0 0 0 588.4 75.6L560.1 73c-4.7-.4-8.7 3.2-8.7 7.9v383.7c0 4.4 3.6 8 8 8l383.6-1c4.7-.1 8.4-4 8-8.6z"
          />
        </svg>
      </span>
      <p>
        Your Progress
      </p>
    </div>
  </button>
  <button
    onClick={[Function]}
  >
    <div>
      <span>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M5 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm14 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-7 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
            />
          </g>
        </svg>
      </span>
      <p>
        More
      </p>
    </div>
  </button>
</div>
`;
